setup.py

```diff
@@ -0,0 +1,149 @@
+            '-DCMAKE_INSTALL_RPATH=$ORIGIN',
+            '-DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON',
+            '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF',
+            # Windows: will already have %PATH% in package dir
+        ]
+
+        cfg = 'Debug' if self.debug else 'Release'
+        build_args = ['--config', cfg]
+
+        if platform.system() == "Windows":
+            cmake_args += [
+                '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(
+                    cfg.upper(),
+                    extdir
+                )
+            ]
+            if sys.maxsize > 2**32:
+                cmake_args += ['-A', 'x64']
+            build_args += ['--', '/m']
+        else:
+            cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
+            build_args += ['--', '-j2']
+
+        env = os.environ.copy()
+        env['CXXFLAGS'] = '{} -DVERSION_INFO=\\"{}\\"'.format(
+            env.get('CXXFLAGS', ''),
+            self.distribution.get_version()
+        )
+        if not os.path.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+        subprocess.check_call(
+            ['cmake', ext.sourcedir] + cmake_args,
+            cwd=self.build_temp,
+            env=env
+        )
+        subprocess.check_call(
+            ['cmake', '--build', '.'] + build_args,
+            cwd=self.build_temp
+        )
+
+with open('./README.md', encoding='utf-8') as f:
+    long_description = f.read()
+
+# Get the package requirements from the requirements.txt file
+with open('./requirements.txt') as f:
+    install_requires = [line.strip('\n') for line in f.readlines()]
+
+setup(
+    name='openPMD-api',
+    version='0.6.2a',
+    author='Fabian Koller, Axel Huebl',
+    author_email='f.koller@hzdr.de, a.huebl@hzdr.de',
+    maintainer='Axel Huebl',
+    maintainer_email='a.huebl@hzdr.de',
+    description='C++ & Python API for Scientific I/O with openPMD',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    url='https://github.com/openPMD/openPMD-api.git',
+    license='LGPL-3.0',
+    ext_modules=[CMakeExtension('openPMD')],
+    cmdclass=dict(build_ext=CMakeBuild),
+    zip_safe=False,
+    # tests_require=['pytest'],
+    install_requires=install_requires,
+    # extras_require = {
+    #    'GUI':  ["ipywidgets", "matplotlib", "cython"],
+    #    'plot': ["matplotlib", "cython"],
+    #    'tutorials': ["ipywidgets", "matplotlib", "wget", "cython"]
+    # },
+    # cmdclass={'test': PyTest},
+    # platforms='any',
+    classifiers=[
+        'Development Status :: 3 - Alpha',
+        'Natural Language :: English',
+        'Environment :: Console',
+        'Intended Audience :: Science/Research',
+        'Operating System :: OS Independent',
+        'Topic :: Scientific/Engineering',
+        'Topic :: Database :: Front-Ends',
+        'Programming Language :: C++',
+        'Programming Language :: Python',
```

Comment:
this line probably needs to go to make sure it's really just matching Python 3 on PyPI

Reply:
uh, it's [`python_requires`](https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires) for setuptools on PyPI.
([ref](https://github.com/pypa/pip/issues/3182#issuecomment-289073930))